### PR TITLE
fix(v2): refactor icons in theme-classic, fix swizzle issue

### LINKED ISSUE #3830: When DocSidebar is swizzled, arrow.svg  is missing
## 🐛 Bug Report

When DocSidebar is swizzled, a `ModuleNotFoundError` occurs due to `arrow.svg` missing. 

### Have you read the Contributing Guidelines on issues

Yes.

## To Reproduce

1. Initialize a new project by running:
   ```   
   npx @docusaurus/init@latest init my-website classic
   ```

1. Swizzle the DocSidebar component:
   ```
   npm run swizzle @docusaurus/theme-classic DocSidebar
   ```

1. Run docusaurus:
   ```
   npm start
   ```

## Expected behavior

No errors should occur.

## Actual Behavior

The following error appears in the terminal:

```bash
./src/theme/DocSidebar/styles.module.css
ModuleNotFoundError: Module not found: Error: Can't resolve '../icons/arrow.svg' in 'C:\temp\my-website\src\theme\DocSidebar'
 @ ./src/theme/DocSidebar/index.js 6:541-581 9:634-640 9:2413-2419 9:2430-2436 9:2478-2484 9:2587-2593 9:2706-2712 9:2755-2761 9:3172-3178 9:3195-3201 9:3303-3309 9:4133-4139
 @ ./node_modules/@docusaurus/theme-classic/lib/theme/DocPage/index.js
 @ ./.docusaurus/registry.js
 @ ./node_modules/@docusaurus/core/lib/client/exports/ComponentCreator.js
 @ ./.docusaurus/routes.js
 @ ./node_modules/@docusaurus/core/lib/client/clientEntry.js
 @ multi ./node_modules/react-dev-utils/webpackHotDevClient.js ./node_modules/@docusaurus/core/lib/client/clientEntry.js
```

## Your Environment

- Docusaurus version used: 2.0.0-alpha.69

### PR DESCRIPTION
**Pull request by Simek**
## Motivation

Fixes #3830 

This PR refactor all SVG usages in the `theme-classic` and introduces three new and swizzable icon components:
* `IconArrow`,
* `IconEdit`,
* `IconMenu`.

I have tried to add define them in the `icons` folder but there were some wired import issues on build, maybe some one will be able to improve the directory structure in the future.

This refactor also fixes the swizzle problem, which occurs currently for `DocPage` and `DocSidebar`.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Local run of V2 Docusuaurs website with swizzled `DocSidebar` component.

## Related PRs

* #3840 - after this PR merge the CI Swizzle, after rebase,, should also be successful.

### CODE REVIEW

File: packages/docusaurus-theme-classic/src/theme/IconEdit/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      fill="currentColor"
+      height="1.2em"
+      width="1.2em"
+      preserveAspectRatio="xMidYMid meet"
+      role="img"
+      viewBox="0 0 40 40"
+      style={{
+        marginRight: '0.3em',
+        verticalAlign: 'sub',
+      }}
```
Comment (slorber) [inline packages/docusaurus-theme-classic/src/theme/IconEdit/index.tsx:23]: "external" styles should rather be passed as props and keep in the parent component.

Some refs:
https://visly.app/blogposts/layout-isolated-components
https://mxstbr.com/thoughts/margin

**Review by slorber — CHANGES_REQUESTED** [review]
thanks, that's a helpful refactor that we'd probably have to do to create the theme gallery anyway 👍 

just I'd rather not hardcode a margin inside the SVG directly

**Comment by Simek** [comment]
> just I'd rather not hardcode a margin inside the SVG directly

It was how the inline SVG code looked like before refactor. 

I will extract the inilne styles in the next commit.

**Comment by slorber** [comment]
it was fine before because the svg was inlined, if we make a real comp we should only put "internal styles" in it, so that the component can be used safely in multiple different contexts.

**Comment by slorber** [comment]
Thanks

Actually I meant the styles like margin should not be at all in the IconEdit comp, but in the comps using them (ie `<IconEdit className={styles.docEditButton}/>`)

But in this case, it would create some duplication as the  comp is used in the same way from 2 places, so let's merge and see 🤷‍♂️

**Pull request merged**